SYSTEMD: chown() sssd.conf in service file

alexey-tikhonov opened the pull request.

jakub-vavra-cz commented on src/sysv/systemd/sssd.service.in:
```diff
@@ -10,6 +10,9 @@ StartLimitBurst=5
 EnvironmentFile=-@environment_file@
+ExecStartPre=+-/bin/chown -f @SSSD_USER@:@SSSD_USER@ /etc/sssd/sssd.conf
+ExecStartPre=+-/bin/chown -f -R @SSSD_USER@:@SSSD_USER@ /etc/sssd/conf.d
+ExecStartPre=+-/bin/chown -f -R @SSSD_USER@:@SSSD_USER@ /etc/sssd/pki
```
Might be worth adding the /var/log/sssd as well just for a good measure.

alexey-tikhonov replied:
On the other hand, logs (and /var/lib/sss/db/* content) is different vs 'sssd.conf'

'sssd.conf' can be created by a 3rd party (realm, ansible script, etc)
logs and ldb-cache, on the other hand, are only created by SSSD itself.

I think it's reasonable to expect that in a rare case of re-configuration of SSSD service user, admin should wipe those artifacts as well. So doesn't justify additional "permanent" lines in service description, and I rather tend to remove corresponding lines from socket activated services.

jakub-vavra-cz replied:
Ack. We just need to consider updates and have appropriate pieces in scriptlets to be able to upgrade/rollback existing configuration including  /var/lib/sss/db/ and logs (with the directory).

alexey-tikhonov replied:
https://github.com/SSSD/sssd/blob/ca684cd156debbdf3d95776271e7020b883b1e81/contrib/sssd.spec.in#L1111 ?

ikerexxe reviewed (approved):
I reviewed the last two commits and they look good to me.

alexey-tikhonov commented on src/sysv/systemd/sssd-kcm.service.in:
```diff
@@ -9,6 +9,8 @@ Also=sssd-kcm.socket
 
 [Service]
 Environment=DEBUG_LOGGER=--logger=files
+ExecStartPre=+-/bin/chown -f @SSSD_USER@:@SSSD_USER@ /etc/sssd/sssd.conf
```
Fixed

sumit-bose commented:
Hi,

in general I'm fine with the changes. It might be worth to add a `:relnote:` to the patch which removed the chown for the logs files to let e.g. packages know that it is expected that chown happens during the package update if the user SSSD is run under should be changed.

bye,
Sumit

alexey-tikhonov commented:
> It might be worth to add a `:relnote:` to the patch which removed the chown for the logs files

Added (and rebased).

But most probably all notes on "non privileged SSSD" topic will have to be reviewed / consolidated / edited before a release anyway.

justin-stephenson reviewed (approved):
Ack.

sumit-bose reviewed (approved):
Hi,

thanks for the updates, ACK.

bye,
Sumit

ikerexxe reviewed (approved):
I took another glance and it looks good to me.

alexey-tikhonov commented:
Pushed PR: https://github.com/SSSD/sssd/pull/7334

* `master`
    * 5531e1de5be2a5b0be1f016e36d5a4ee8a551845 - SYSTEMD: don't chown() logs
    * 7bab23612584bc5e7a6380f9eb0dfd18e63df8e5 - SYSTEMD: chown() sssd.conf in service file
